samples/downloadPublicFile.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//   title: Storage Download Public File.
+//   description: Download Public File.
+//   usage: node downloadPublicFile.js <BUCKET_NAME> <SRC_FILE_NAME> <DEST_FILE_NAME>
+
+const path = require('path');
+const cwd = path.join(__dirname, '..');
+
+function main(
+  bucketName = 'my-bucket',
+  srcFileName = 'test.txt',
+  destFileName = path.join(cwd, 'downloaded.txt')
+) {
+  // [START storage_download_public_file]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const bucketName = 'Name of a bucket, e.g. my-bucket';
+  // const srcFileName = 'Remote file to download, e.g. file.txt';
+  // const destFileName = 'Local destination for file, e.g. ./local/path/to/file.txt';
+
+  // CI authentication is done with ADC. Cache it here, restore it after file downloaded.
+  const GOOGLE_APPLICATION_CREDENTIALS =
```

Comment:
Is there no way to pass in a config with no credentials?

Working with environment variables is definitely something we shouldn't do.